doc/MSTP/MSTP.md

```diff
@@ -0,0 +1,1148 @@
+1. Verify CLI to clear MSTP region revision. 
+1. Verify CLI to clear MSTP port priority 
+1. Verify CLI to clear MSTP hello-time 
+1. Verify CLI to clear MSTP cost-value 
+1. Verify CLI to clear MSTP max-hops 
+1. Verify CLI to clear MSTP root guard 
+1. Verify CLI to clear MSTP bpdu guard 
+1. Verify CLI to clear bridge priority in an instance. 
+1. Verify CLI to map VLAN to an instance 
+1. Verify CLI to delete VLAN from an instance. 
+1. Verify CLI to set interface priority on a per instance basis. 
+1. Verify CLI to set interface cost on a per instance basis. 
+1. Verify CLI to display all information related to the region. 
+1. Verify CLI to display information related to a specific instance. 
+1. Verify CLI to display information about a specific interface in a specific instance. 
+1. Verify CLI to display statistics. 
+1. Verify CLI to display statistics on a per instance basis. 
+1. Verify CLI to clear statistics. 
+1. Verify CLI to clear statistics on a per instance basis. 
+1. Verify CLI to clear statistics on a per interface per instance basis. 
+1. Verify the commands that are disabled for MSTP mode 
+
+
+### Functional Test Cases 
+
+1. Verify CONFIG DB is populated with configured MSTP parameters. 
+1. Verify MSTP instances running on multiple VLANs. 
+1. Verify multiple VLANs can be mapped to a single instance. 
+1. Verify VLAN to STP instance mapping is populated correctly in APP DB, ASIC DB, Hardware 
+1. Verify instance STP ports are created for its VLAN members. 
+1. Verify correct format of MSTP BPDU. 
+1. Verify MSTP traps are created. 
+1. Verify correct operation of CIST and MSTIs. 
+1. Verify two bridges are in the same region only if they have the same region name, revision and instance to VLAN mapping. 
+1. Verify the loop free topology inside and between regions. 
+1. Verify the operational values of forward delay, max age, hello timer and max hops are of the CIST root. 
+1. Verify FDB flush as a result of topology change. 
+1. Verify instance is active only when there is at least one VLAN member of any of the VLANs that are mapped to it. 
+1. Verify altering bridge priority can alter the selection of CIST root, regional root and MSTI root. 
+
+1. Verify altering port priority can alter the selection of designated port. 
+1. Verify max-hops by changing value. 
+
+1. Verify hello-timer by changing timer values 
+
+1. Verify max-age by changing intervals 
+
+1. Verify altering bridge priority will alter Root Bridge selection 
+
+1. Verify altering port priority will alter Root port selection 
+
+1. Verify altering port priority will alter Designated port selection 
+
+1. Verify altering port cost results in path with lowest root path cost is seleced as root port 
+
+1. Verify port states on same physical interface 
+
+1. Verify MSTP interoperability with STP, RSTP. 
+
+1. Verify MSTP operational data is synced to APP DB and ASIC DB correctly. 
+
+1. Verify MSTP over LAG. 
+
+1. Verify MSTP over static breakout ports. 
```

Comment:
Can we add test-cases for BPDU guard and other functionalities?

Reply:
Added more test-cases for BPDU guard and other functionalities